src/client/shader.cpp

```diff
@@ -462,7 +462,7 @@ u32 ShaderSource::getShaderIdDirect(const std::string &name,
 	infostream<<"getShaderIdDirect(): "
 			<<"Returning id="<<id<<" for name \""<<name<<"\""<<std::endl;
 
-	return id;
+	return info.shader_failed ? 0 : id;
```

Comment:
are you sure 0 can't be a valid shader id?

Reply:
The rest of the function seems to return 0 on failure https://github.com/minetest/minetest/blob/fc083643cb7f4a6697c6a45d70c0dbe6324b555c/src/client/shader.cpp#L427